zkevm-circuits/src/evm_circuit/execution/error_oog_dynamic_memory.rs

```diff
@@ -0,0 +1,365 @@
+            "expanded address overflow {} bytes",
+            N_BYTES_MEMORY_ADDRESS
+        );
+        self.expanded_address_in_range.assign(
+            region,
+            offset,
+            F::from(expanded_address),
+            F::from(1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
+        )?;
+
+        self.memory_expansion.assign(
+            region,
+            offset,
+            step.memory_word_size(),
+            [expanded_address],
+        )?;
+
+        self.insufficient_gas.assign(
+            region,
+            offset,
+            F::from(step.gas_left),
+            F::from(step.gas_cost),
+        )?;
+
+        self.rw_counter_end_of_reversion.assign(
+            region,
+            offset,
+            Value::known(F::from(call.rw_counter_end_of_reversion as u64)),
+        )?;
+
+        self.restore_context.assign(
+            region,
+            offset,
+            block,
+            call,
+            step,
+            if opcode == OpcodeId::CREATE { 5 } else { 4 },
+        )?;
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::test_util::CircuitTestBuilder;
+    use eth_types::{bytecode, word, Bytecode, ToWord};
+    use mock::test_ctx::helpers::account_0_code_account_1_no_code;
+    use mock::{eth, TestContext, MOCK_ACCOUNTS};
+
+    #[test]
+    fn test() {
+        let codes = [
+            bytecode! {
+                PUSH8(0xFF)
+                PUSH32(word!("0xffffffff")) // offset
+                RETURN
+                STOP
+            },
+            bytecode! {
+                PUSH8(0xFF)
+                PUSH32(word!("0xffffffff")) // offset
+                REVERT
+                STOP
+            },
+            bytecode! {
+                PUSH8(0xFF)
+                PUSH32(word!("0xffffffff")) // offset
+                PUSH8(0x0) // value
+                CREATE
+                STOP
+            },
+        ];
+
+        for code in codes.iter() {
+            test_root(code.clone());
+            test_internal(code.clone());
+        }
+    }
+
+    fn test_root(code: Bytecode) {
+        let ctx = TestContext::<2, 1>::new(
+            None,
+            account_0_code_account_1_no_code(code),
+            |mut txs, accs| {
+                txs[0]
+                    .from(accs[1].address)
+                    .to(accs[0].address)
+                    .gas(word!("0xFFFF"));
+            },
+            |block, _tx| block,
+        )
+        .unwrap();
+
+        CircuitTestBuilder::new_from_test_ctx(ctx).run();
+    }
+
+    fn test_internal(code: Bytecode) {
+        let code_a = bytecode! {
+            PUSH1(0x00) // retLength
+            PUSH1(0x00) // retOffset
+            PUSH32(0x00) // argsLength
+            PUSH32(0x00) // argsOffset
+            PUSH1(0x00) // value
+            PUSH32(MOCK_ACCOUNTS[1].to_word()) // addr
+            // Decrease expected gas cost (by 1) to trigger out of gas error.
+            PUSH32(0xFFFF) // gas
```

Comment:
make sure gas provided can cause oog error.

Reply:
that's a copy paste problem..